companies/services/financial_metrics.py

```
@@ -184,12 +267,80 @@ def calculate_all_metrics(self, financial_statement: FinancialStatement) -> dict
+                    and previous_statement.operating_profit
+                ):
+                    yoy_operating_profit = self.calculate_yoy_growth(
+                        financial_statement.operating_profit,
+                        previous_statement.operating_profit,
+                    )
+
+                if financial_statement.net_income and previous_statement.net_income:
+                    yoy_net_income = self.calculate_yoy_growth(
+                        financial_statement.net_income, previous_statement.net_income
+                    )
+        except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)